src/control/Image/Image.js

```diff
@@ -0,0 +1,143 @@
+    print.title = 'Print image';
+    print.className = 'print';
+    print.addEventListener('click', this.printImage.bind(this));
+    element.appendChild(print);
+  }
+
+  /**
+   * Callback to deactivate the image control.
+   * @private
+   */
+  deactivate() {
+    this.element.classList.remove('active');
+  }
+
+  /**
+   * Callback for the image button click event.
+   * @param {MouseEvent} event The event to handle
+   * @private
+   */
+  captureImage(event) {
+    event.preventDefault();
+
+    // Create a new canvas element to combine multiple map canvas data to.
+    const mapCanvas = document.createElement('canvas');
+    const [width, height] = this.getMap().getSize();
+    mapCanvas.width = width;
+    mapCanvas.height = height;
+    const mapContext = mapCanvas.getContext('2d');
+
+    // Draw each canvas from this map into the new canvas.
+    Array.prototype.forEach.call(
+      this.getMap().getTargetElement().querySelectorAll('.ol-layer canvas'),
+      (canvas) => {
+        if (canvas.width > 0) {
+          const { opacity } = canvas.parentNode.style;
+          mapContext.globalAlpha = opacity === '' ? 1 : Number(opacity);
+
+          // Get the transform parameters from the style's transform matrix.
+          // This is necessary so that vectors align with raster layers.
+          const { transform } = canvas.style;
+          const matrix = transform
+            .match(/^matrix\(([^(]*)\)$/)[1]
+            .split(',')
+            .map(Number);
+
+          // Apply the transform to the export map context.
+          CanvasRenderingContext2D.prototype.setTransform.apply(
+            mapContext,
+            matrix,
+          );
+          mapContext.drawImage(canvas, 0, 0);
+        }
+      },
+    );
+
+    // Build a jpeg data url and update link.
+    const url = mapCanvas.toDataURL('image/jpeg');
+    this.link.href = url;
+
+    // Remove the new canvas.
+    mapCanvas.remove();
+
+    // Toggle the image actions.
+    this.element.classList.add('active');
+
+    // Subscribe to events to deactivate image actions.
+    this.getMap().on(EventType.CLICK, this.deactivate.bind(this));
+    this.getMap().on(EventType.CHANGE, this.deactivate.bind(this));
+    this.getMap().on(EventType.MOVESTART, this.deactivate.bind(this));
+
+    // The move start event seems to only work with a string.
+    this.getMap().on('movestart', this.deactivate.bind(this));
```

Comment:
That is because `EventType.MOVESTART` doesn't exist: https://github.com/openlayers/openlayers/blob/v6.14.1/src/ol/events/EventType.js#L24-L38

You're looking for `MapEventType.MOVESTART`: https://github.com/openlayers/openlayers/blob/v6.14.1/src/ol/MapEventType.js#L21

Reply:
Ahhh good catch! My IDE was getting confused with these types... but I believe I got it imported correctly.